homework/App.js

```diff
@@ -7,58 +7,50 @@ class App {
     this.initialize(url);
   }
 
-  /**
-   * Initialization
-   * @param {string} url The GitHub URL for obtaining the organization's repositories.
-   */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
+    const imgContainer = Util.createAndAppend('div', root, { class: 'logo' });
+    Util.createAndAppend('img', imgContainer, {
+      src: './hyf.png',
+      id: 'hyf-logo',
+      alt: 'logo image',
+    });
+    const header = Util.createAndAppend('header', root, { class: 'header' });
+    Util.createAndAppend('p', header, { html: 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, {
+      class: 'selectBox',
+      'aria-label': 'HYF Repositories',
+    });
+    Util.createAndAppend('div', root, { id: 'container' });
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const fetch = await Util.fetchJSON(url);
```

Comment:
This name `fetch` confused me for the following reasons:

1. `fetch` is also the name of the standard browser function `fetch()`: this is a replacement for `XMLHttpRequest` and directly works with promises.
2. The name is a verb and verbs should be reserved for functions (functions perform actions). Variables should use nouns (they reference _things_).
3. The name does not reflect the contents: an array of repository objects.

In this case even the name `data` or `repositoryData` would be better than `fetch`.

Reply:
yes, you are right, I have to improve in naming everything, I just used fetch because i am awaiting the fetchJSON, no other reasons, I will change it later xd... I am not so creative with naming but it will come better with time! haha